File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        
        for (int n : nums) {
            set.add(n);
        }
        
        int max = nums.length / 2;
        
        if(max < set.size()) {
            return max;
        } else {
            return set.size();
        }
    
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> nums)
{
    set<int> poketmon;
    
    for(int num : nums) {
        poketmon.insert(num);
    }
    
    return poketmon.size() >= nums.size() / 2 ? nums.size() / 2 : poketmon.size();
}